src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs

```diff
@@ -191,6 +192,48 @@ public static Scope CreateDbCommandScope(Tracer tracer, IDbCommand command)
             return scope;
         }
 
+        public static Scope CreateRabbitMQScope(Tracer tracer, out RabbitMQTags tags, string command, ISpanContext parentContext = null, string spanKind = SpanKinds.Client, DateTimeOffset? startTime = null, string queue = null, string exchange = null, string routingKey = null)
+        {
+            tags = null;
+
+            if (!tracer.Settings.IsIntegrationEnabled(RabbitMQConstants.IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                Span parent = tracer.ActiveScope?.Span;
+
+                tags = new RabbitMQTags(spanKind);
+                string serviceName = tracer.Settings.GetServiceName(tracer, RabbitMQConstants.ServiceName);
+                scope = tracer.StartActiveWithTags(RabbitMQConstants.OperationName, parent: parentContext, tags: tags, serviceName: serviceName, startTime: startTime);
+                var span = scope.Span;
+
+                span.Type = SpanTypes.Queue;
+                span.ResourceName = command;
+                tags.Command = command;
+
+                tags.Queue = queue;
+                tags.Exchange = exchange;
+                tags.RoutingKey = routingKey;
+
+                tags.InstrumentationName = RabbitMQConstants.IntegrationName;
+                tags.SetAnalyticsSampleRate(RabbitMQConstants.IntegrationId, tracer.Settings, enabledWithGlobalSetting: false);
```

Comment:
I think we want `enabledWithGlobalSetting: true` if `spanKind == SpanKinds.Server`.

Reply:
Does this apply for queueing technologies? I can add it if `spanKind == SpanKinds.Consumer` but I still don't fully understand what the effect of this change is

Reply:
Never mind. I checked our RFC and `false` is the correct setting. We should only enable analytics with the global setting in web frameworks spans (aspnet, mvc, webapi, etc). Resolving this comment.